src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1492,28 +1497,42 @@
           (select-keys migrations pending))))
 
 (defn migrate!
-  "Migrates database to the latest schema version. Does nothing if database is
-  already at the latest schema version."
-  []
+  "Migrates database to the latest schema version. Does nothing if
+  database is already at the latest schema version.  Requires a
+  connection pool because some operations may require an indepdendent
+  database connection."
+  [db-connection-pool]
   (if-let [unexpected (first (difference (applied-migrations) (kitchensink/keyset migrations)))]
     (throw (IllegalStateException.
             (format "Your PuppetDB database contains a schema migration numbered %d, but this version of PuppetDB does not recognize that version."
                     unexpected))))
-
-  (if-let [pending (seq (pending-migrations))]
-    (sql/transaction
-     (doseq [[version migration] pending]
-       (log/infof "Applying database migration version %d" version)
-       (try
-         (migration)
-         (record-migration! version)
-         (catch java.sql.SQLException e
-           (log/error e "Caught SQLException during migration")
-           (let [next (.getNextException e)]
-             (when-not (nil? next)
-               (log/error next "Unravelled exception")))
-           (System/exit 1)))))
-    (log/info "There are no pending migrations")))
+  (letfn [(sql-or-die [f]
+            (try
+              (f)
+              (catch java.sql.SQLException e
+                (log/error e "Caught SQLException during migration")
+                (let [next (.getNextException e)]
+                  (when-not (nil? next)
+                    (log/error next "Unravelled exception")))
+                ;; See PDB-1118
```

Comment:
https://tickets.puppetlabs.com/browse/PDB-1118
Not sure I understand what you mean by referring to this issue. I'd also rather see whatever you want to say written out here; issue trackers come and go, while source code has many lives.